Source/WebInspectorUI/UserInterface/Views/OverrideDeviceSettingsPopover.js

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+WI.OverrideDeviceSettingsPopover = class OverrideDeviceSettingsPopover extends WI.SettingsPopover
+{
+    constructor(delegate)
+    {
+        super(delegate);
+    }
+
+    // Protected
+
+    createContentElement()
+    {
+        let contentElement = document.createElement("div");
+        contentElement.classList.add("device-settings-content");
+
+        if (InspectorFrontendHost.isRemote)
+            contentElement.appendChild(this._createUserAgentSection());
+
+        if (WI.DeviceSettingsManager.supportsSetScreenSizeOverride())
+            contentElement.appendChild(this._createScreenSizeSection());
+
+        contentElement.appendChild(this._createSettingsSection());
+        contentElement.appendChild(WI.ReferencePage.DeviceSettings.createLinkElement());
+
+        return contentElement;
+    }
+
+    // Private
+
+    _createSettingsSection()
+    {
+        let coreSettingsGroup = {
+            name: WI.UIString("Core Features"),
```

Comment:
IMO this is kinda a weird name.  How about "General" (and `generalSettingsGroup`)?

Reply:
We resolved in a weekly meeting about these features to use the term "Core Features" for these three fundamental features of the web platform. IMO it helps drive the impact these three feature have, or rather their absence will have. I'm not opposed to "General", but it would be a divergence from our current plan of record.

Reply:
i dunno IMO the name "Core Features" just feels really wrong

it almost implies that the other things arent "core" which i dont think is all that accurate (e.g. cross origin stuff is pretty core to the web (especially nowadays)).  it also doesnt really match what we've done elsewhere in Web Inspector (e.g. the General pane of the Settings Tab)

Reply:
I'm not opposed to "General" either, but I'll defer to Patrick who has discussed the details of this redesign with other people.

Reply:
General settings should typically be things that are relatively safe/non-destructive to toggle. These three settings really aren't that 😅. "Core Features" communicates the impact turning off any of these three settings will have. The web continues to behave as a user would likely expect it to if you disable something like CORS (ya know, minus the security issues). But when you turn one of these off, the web becomes noticeably different, hence the "core"-ness of these three features. However, this isn't a point worth us getting stuck in review for in either direction.